python/lsst/rapid/analysis/quickLook.py

```diff
@@ -0,0 +1,281 @@
+            Result struct with component:
+            - ``exposure`` : `afw.image.Exposure`
+                The ISRed and cosmic-ray-repaired exposure.
+
+        """
+        isrConfig = isrBaseConfig if isrBaseConfig else IsrTask.ConfigClass()
+        isrConfig.doBias = False
+        isrConfig.doDark = False
+        isrConfig.doFlat = False
+        isrConfig.doFringe = False
+        isrConfig.doDefect = False
+        isrConfig.doLinearize = False
+        isrConfig.doCrosstalk = False
+        isrConfig.doBrighterFatter = False
+        isrConfig.usePtcGains = False
+
+        if bias:
+            isrConfig.doBias = True
+            self.log.info("Running with bias correction")
+
+        if dark:
+            isrConfig.doDark = True
+            self.log.info("Running with dark correction")
+
+        if flat:
+            isrConfig.doFlat = True
+            self.log.info("Running with flat correction")
+
+        # TODO: deal with fringes here
+        if defects:
+            isrConfig.doDefect = True
+            self.log.info("Running with defect correction")
+
+        if linearizer:
+            isrConfig.doLinearize = True
+            self.log.info("Running with linearity correction")
+
+        if crosstalk:
+            isrConfig.doCrosstalk = True
+            self.log.info("Running with crosstalk correction")
+
+        if bfKernel:
+            isrConfig.doBrighterFatter = True
+            self.log.info("Running with brighter-fatter correction")
+
+        if ptc:
+            isrConfig.usePtcGains = True
+            self.log.info("Running with ptc correction")
+
+        isrConfig.doWrite = False
+        isrTask = IsrTask(config=isrConfig)
+        result = isrTask.run(ccdExposure,
+                             camera=camera,
+                             bias=bias,
+                             dark=dark,
+                             flat=flat,
+                             #  fringes=pipeBase.Struct(fringes=None),
+                             defects=defects,
+                             linearizer=linearizer,
+                             crosstalk=crosstalk,
+                             bfKernel=bfKernel,
+                             bfGains=bfGains,
+                             ptc=ptc,
+                             crosstalkSources=crosstalkSources,
+                             isGen3=isGen3,)
+
+        postIsr = result.exposure
+
+        if self.config.doRepairCosmics:
+            try:  # can fail due to too many CRs detected, and we always want an exposure back
+                self.log.info("Repairing cosmics...")
+                if postIsr.getPsf() is None:
+                    installPsfTask = InstallGaussianPsfTask()
+                    installPsfTask.run(postIsr)
+
+                repairConfig = CharacterizeImageTask.ConfigClass()
+                repairConfig.doMeasurePsf = False
+                repairConfig.doApCorr = False
+                repairConfig.doDeblend = False
```

Comment:
I suspect you also want `doWrite=False` here?

I'm guessing you're not just using `RepairTask` here--instead of Characterize with most things disabled--is because you want the single iteration of `detectMeasureAndEstimatePsf` that happens there? This feels a bit fragile, as Characterize could change and you'd have to modify this to disable other things. I'm not sure if there is a better approach.

Reply:
Definitely do want `doWrite=False` there, yes, thanks - added.

Reply:
As for the iteration, I bet I could make do without actually - I suspect manually installing a nice fat SimpleGaussianPsf or whatever it's called would be easily good enough here, especially given AuxTel's platescale (though I would like this particular thing not to fall down the AuxTel-only hole). I'll give that a shot once I've done all the other comments, but I think that should perhaps just be another ticket in the name of convergance.

Reply:
I've added to TODO here to try this out in future.